src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -57,7 +68,67 @@ protected AssemblyLoadContext()
         internal AssemblyLoadContext(bool fDummy)
         {
         }
-        
+#else // !FEATURE_COLLECTIBLE_ALC
+        [System.Security.SecuritySafeCritical]
+        protected AssemblyLoadContext()
+            : this(true) // TODO: Testing only, should default to false!
+        {
+        }
+
+        protected AssemblyLoadContext(bool isCollectible)
+        {
+            // Initialize the VM side of AssemblyLoadContext if not already done.
+            GCHandle gchALC = GCHandle.Alloc(this, isCollectible ? GCHandleType.WeakTrackResurrection : GCHandleType.Normal);
+            IntPtr ptrALC = GCHandle.ToIntPtr(gchALC);
+            
+            m_pNativeAssemblyLoadContext = InitializeAssemblyLoadContext(ptrALC, isCollectible);
+
+            m_isCollectible = isCollectible;
+            m_gchALC = gchALC;
+        }
+
+        ~AssemblyLoadContext()
+        {
+            if (!m_isCollectible)
```

Comment:
GCHandle for ALC is created on the managed side and shared with the VM (via InitiaizeAssemblyLoadContext). Once the VM has destroyed/released the native side of ALC, it is reasonable for the AssemblyLoadContext instance to be collectible and have its finalizer run.